cityofaustin/atd-moped
# Various additional fields for ATSD AGOL view

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/17606

This PR adds four additional columns to the `component_arcgis_online_view` DB view:
- `council_districts_searchable` (makes council districts more searchable in ArcGIS)
- `substantial_completion_date_estimated` (see explanation in issue and test steps below)
- `related_project_ids` (shows all subproject ids in a text list)
- `related_project_ids_searchable` (shows all subproject ids in a list that is more searchable in ArcGIS)

## Testing
**URL to test:** 

Local

**Steps to test:**
1. Start up the local stack with production data. Create a new project and add a component that spans multiple council disctricts. I created a bike lane that goes east to west from Neches to San Marcos on 7th St. Note the component id in the url for future DB queries.
2. Open up your database client and run the following query:
```sql
SELECT
	council_districts,
	council_districts_searchable
FROM
	component_arcgis_online_view
WHERE
	project_component_id = <your component id>;
```
3. You should see that single digit districts have one zero padding to the left in the new searchable version of the column.
4. Now, add three subprojects to your project in the project summary view. Then, test the following query:
```sql
SELECT
	related_project_ids,
	related_project_ids_searchable
FROM
	component_arcgis_online_view
WHERE
	project_component_id = <your component id>;
```
5. You should see a list of the subproject ids in related project ids column and a list of related project ids with left-padded zeroes going up to 5 digits in the searchable columns. These lists should include the project you created (parent of the child projects).
6. Next, delete all phases from your project in the **Timeline** tab. Add a phase that is either **Complete** or **Post-construction** with a start date that is before today with the **Confirmed** checkbox unchecked. Save.
7. Check the `substantial_completion_date_estimated` with the following query:
```sql
SELECT
	substantial_completion_date, substantial_completion_date_estimated
FROM
	component_arcgis_online_view
WHERE
	project_component_id = <your component id>
```
8. The `substantial_completion_date_estimated` should contain the date of your new phase. The `substantial_completion_date` should be null.
9. Add another phase that is either **Complete** or **Post-construction** with a end date earlier than your last phase and the **Confirmed** checkbox unchecked. The `substantial_completion_date_estimated` should have the date of your earliest phase start or end date and `substantial_completion_date` should still be null.
10. Add another phase that is either **Complete** or **Post-construction** with a start or end date before today and leave the **Confirmed** checkbox checked. Now, `substantial_completion_date` should have your confirmed date and `substantial_completion_date_estimated` should be null. 


---
#### Ship list
